### PR TITLE
Fix challenge hash and add test code to verify signatures against libbitcoin_consensus

### PR DESCRIPTION
**Pull request by xoloki**
While exploring auditor recommendations regarding double hashing of tags, it became apparent that the `XMD` expansion PR changed the format of the `WSTS` signature challenge hash, such that it was no longer `taproot` compliant.  This would have been caught during PR review if we had a test that verified `WSTS` signatures against bitcoin consensus.

So this PR takes test code from `sBTC` which does such validation, and uses it to verify that `WSTS` signers produce valid signatures.

### CODE REVIEW

File: src/btc.rs [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    #[error("error returned from libbitcoinconsensus: {0}")]
+    BitcoinConsensus(bitcoinconsensus::Error),
+    /// Taproot error
+    #[error("an error occurred when constructing the taproot signing digest: {0}")]
+    Taproot(#[from] bitcoin::sighash::TaprootError),
+}
+
+/// An unspent transaction output, which contains all of the information needed to identify or spend
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct UTxO {
```
Comment (djordon) [inline src/btc.rs]: nit: stylistically, this should probably be `Utxo` to match the usual Rust [naming conventions](https://rust-lang.github.io/api-guidelines/naming.html). But I'm surprised that clippy didn't complain, do we run clippy on test code?

File: src/btc.rs [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+
+    #[test]
+    fn verify_sig_no_merkle_root() {
+        verify_sig(None)
+    }
+
+    #[test]
+    fn verify_sig_some_merkle_root() {
+        // XXX key spends with a merkle root are not working
+        //verify_sig(Some([0u8; 32]))
```
Comment (djordon) [inline src/btc.rs:204]: Uh oh

**Review by djordon — COMMENTED** [review]
Had one small nit.

But we have two follow-up pieces of work in this thread, which is to test key spend signing when there is a merkle root in a taproot output, and add a test for script path spending.

File: src/btc.rs (hunk 1 above)
Comment (xoloki) [reply src/btc.rs]: The acronym `UTO` would indeed be `Uto` according to the conventions.  But since the `x` is not a leading letter, `UTxO` seems more appropriate, since it's just a shortened version of `UnspentTransactionOutput`. 

It's not a hill I'm willing to die on, so if you feel strongly enough I'll change it.  But I've always loved the look of `UTxO`!

File: src/btc.rs (hunk 2 above)
Comment (xoloki) [reply src/btc.rs:204]: I don't know the `bitcoin-rs` API well enough to know if I need to do something else to represent a taproot key spend with a script tree.  I suspect I need to add something else, but I don't know of any good example code that shows this.

For the purposes of this PR though, which is to fix the challenge hash and add a test that makes sure it doesn't silently break in the future, fixing key spends with a non-null merkle root seems orthogonal.

File: src/btc.rs (hunk 1 above)
Comment (xoloki) [reply src/btc.rs]: It looks like the enforcement of camel/snake case is done by the compiler, not `clippy`.  But I don't think it groks enough natural language to be able to tell the difference between partial and full acronyms, hence not enforcing `Utxo`.

Comment (xoloki) [reply src/btc.rs]: And interestingly, it looks like `clippy` is *not* being run on test code, we should be passing `--all-targets` which I thought was the default.  I added #175 to address this.

File: src/btc.rs (hunk 2 above)
Comment (djordon) [reply src/btc.rs:204]: Yeah, I comment just to get more information about what happened here, we do not need to fix this in this PR. 

Just to add a quick note here about https://github.com/stacks-sbtc/wsts/issues/173; that ticket is to test both key-path spending when the output contains script-path spending conditions, and script-path spending. As you noted, we currently test key-path spending when there does not exists script-path spending, and this PR fixes a bug there.

File: src/btc.rs (hunk 1 above)
Comment (djordon) [reply src/btc.rs]: We spoke about this at stand-up and I feel like we should rename this to `Utxo`, since that is more likely to follow the naming convention.

Comment (xoloki) [reply src/btc.rs]: Done

**Review by djordon — APPROVED** [review]
Looks good ✅

**Review by jferrant — APPROVED** [review]
